### PR TITLE
Fix inlining for time formatting

### PR DESCRIPTION
**Pull request by shiftylogic**

### CODE REVIEW

**Comment by shiftylogic** [comment]
@ahsonkhan @KrzysztofCwalina

File: src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.Time.cs [hunk 1]
```diff
@@ -133,8 +145,16 @@ static bool TryFormatTimeSpan(TimeSpan value, char format, Span<byte> buffer, ou
...
                 return InvariantUtf16TimeFormatter.TryFormat(value, format, buffer, out bytesWritten);
-            else
-                throw new NotImplementedException();
+
+            ThrowNotImplemented();
+            bytesWritten = 0;
+            return false;
+        }
+
+        // Methods won't be inlined if they contain a throw, so we factor out the throw to a separate method.
```
Comment (ahsonkhan) [inline src/System.Text.Primitives/System/Text/Formatting/PrimitiveFormatter.Time.cs]: Interesting. This is good to know.

**Comment by shiftylogic** [comment]
@dotnet-bot test Innerloop Ubuntu16.04 Release Build and Test

**Comment by KrzysztofCwalina** [comment]
looks good.

**Pull request merged**